backend/compact-connect/lambdas/provider-data-v1/data_model/query_paginator.py

```diff
@@ -0,0 +1,132 @@
+import json
+from base64 import b64decode, b64encode
+from types import MethodType
+from typing import Callable, List
+
+from botocore.exceptions import ClientError
+from marshmallow import ValidationError
+
+from config import config, logger
+from data_model.schema.base_record import BaseRecordSchema
+from exceptions import CCInvalidRequestException, CCInternalException
+
+
+# It's conventional to name a decorator in snake_case, even if it is implemented as a class
+class paginated_query:  # pylint: disable=invalid-name
```

Comment:
This decorator looks really nice. I imagine we can pull it into a common library later so we can share it with other python projects that will have this need.